langserver/references.go

```diff
@@ -296,6 +296,10 @@ func refStreamAndCollect(ctx context.Context, conn jsonrpc2.JSONRPC2, req *jsonr
 				send()
 				return locs
 			}
+			if len(locs) >= limit {
+				stop()
```

Comment:
> just to make sure I read this correct, this limit parameter is just best effort. There may be more than limit number of results returned?

No. We will not send more than limit. This is the goroutine that is responsible for collecting references and sending them. When it calls stop, it is telling the producer it can stop searching for references. If the continue statement just below this line was not there, we could potentially send more found references.

> Is there a reason you chose this over limiting the results on the producer side of the refs channel?

there is one place we consume from the channel, but more than one place we produce. So it is cleaner this way.